projects/plugins/jetpack/modules/videopress/class-videopress-attachment-metadata.php

```diff
@@ -15,27 +15,30 @@ class Videopress_Attachment_Metadata {
 	/**
 	 * Persist the VideoPress metadata information, including rating and display_embed.
 	 *
-	 * @param string|int $post_id       The post id.
-	 * @param string     $guid          VideoPress Guid.
-	 * @param string     $post_title    The post title.
-	 * @param string     $caption       Video caption.
-	 * @param string     $post_excerpt  The post excerpt.
-	 * @param string     $rating        The rating.
-	 * @param int        $display_embed The display_embed.
+	 * @param string|int $post_id        The post id.
+	 * @param string     $guid           VideoPress Guid.
+	 * @param string     $post_title     The post title.
+	 * @param string     $caption        Video caption.
+	 * @param string     $post_excerpt   The post excerpt.
+	 * @param string     $rating         The rating.
+	 * @param int        $display_embed  The display_embed.
+	 * @param int        $allow_download Allow video downloads.
 	 *
 	 * @return bool|\WP_Error
 	 */
-	public static function persist_metadata( $post_id, $guid, $post_title, $caption, $post_excerpt, $rating, $display_embed ) {
+	public static function persist_metadata( $post_id, $guid, $post_title, $caption, $post_excerpt, $rating, $display_embed, $allow_download ) {
 		$post_id = absint( $post_id );
 
 		$args = array(
 			'method'  => 'POST',
 			'headers' => array( 'content-type' => 'application/json' ),
 		);
 
-		$display_embed = (int) $display_embed;
+		// Keep null values to avoid accidental unset 
+		$display_embed  = null === $display_embed ? null : (int) $display_embed;
+		$allow_download = null === $allow_download ? null : (int) $allow_download;
```

Comment:
This fixes an existing bug on Calypso, where checking one of the box would unset the other (or only updating the rating too).